frontend/src/pages/DeptPage.jsx

```diff
@@ -47,9 +48,28 @@ export default function DeptPage({ college, dept, setCourse }) {
   const [classes, error2] = useApi(`/${college}/depts/${dept}/classes`)
   const colleged = campus.find((cmp) => cmp.id === college)
 
-  const cards = courses
-    ? courses.map(({ dept, course, title, classes }) => (
-      <DeptCard
+  const [query, setQuery] = useState('')
+  const [filteredCourses, setFilteredCourses] = useState([])
+  const [filteredClasses, setFilteredClasses] = useState([])
+
+  useEffect(() => {
+    if (courses && classes) {
```

Comment:
Can we separate this so that if one is API is loaded, then the filter can work? No practical impact for us, but might be a good idea.

Reply:
In the new algo I need. both to be present. I can add a fallback, though